streams/src/main/java/org/apache/kafka/streams/state/internals/ExceptionUtils.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.state.internals;
+
+import java.util.LinkedList;
+
+final class ExceptionUtils {
+    private ExceptionUtils() {}
+
+    static LinkedList<RuntimeException> executeAll(final Runnable... actions) {
+        LinkedList<RuntimeException> suppressed = null;
```

Comment:
Why not make it `final` and return an empty list? It seems better to avoid using `null` if possible

Reply:
Just to avoid allocating an empty list in the common case where we don't throw any exceptions.

Reply:
Hmmm... I see your point, but such short living objects should not be much of an overhead? But forgetting to check for `null` caller side is a potential source for bugs?

Reply:
Sounds good. I prefer it the way you suggest anyway. I'll change it.